src/context/AppContext.tsx

```diff
@@ -48,6 +50,14 @@ abstract class BaseAppContextState {
     this.internalIsLoggedIn = newValue;
   }
 
+  public setInternalLoginModalOpen(newValue: boolean) {
+    this.internalLoginModalOpen = newValue;
+  }
+
+  public isInternalLoginModelOpen() {
```

Comment:
This should probably be called `isLoginModalOpen`, as it's supposed to be used from the outside. Also, mod**e**l.